### PR TITLE
Shrink sizeof Value struct

### PR DESCRIPTION
**Pull request by dutor**
* Use pointer instead of inlined value for `std::string`
 * Use bit fields in `DateTime`

### CODE REVIEW

**Comment by Shylock-Hg** [comment]
Great job, take https://github.com/vesoft-inc/nebula-common/pull/488 together.

File: src/common/datatypes/Date.h [hunk 1]
```diff
@@ -129,20 +129,38 @@ inline std::ostream &operator<<(std::ostream& os, const Time& d) {
...
+        struct {
+            uint64_t year:16;
+            uint64_t month:4;
+            uint64_t day:5;
+            uint64_t hour:5;
+            uint64_t minute:6;
+            uint64_t sec:6;
+            uint64_t microsec:22;
+        };
+        uint64_t qword;
```
Comment (critical27) [inline src/common/datatypes/Date.h:142]: 👍

File: src/common/datatypes/Date.h [hunk 2]
```diff
@@ -129,6 +129,10 @@ inline std::ostream &operator<<(std::ostream& os, const Time& d) {
 }
 
 struct DateTime {
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
```
Comment (darionyaphet) [inline src/common/datatypes/Date.h:133]: 👍  it's use to block some warning? would you support some comment?

Comment (dutor) [reply src/common/datatypes/Date.h:133]: sure

File: src/common/datatypes/Date.h [hunk 3]
```diff
@@ -129,20 +129,45 @@ inline std::ostream &operator<<(std::ostream& os, const Time& d) {
...
-    int8_t minute;
-    int8_t sec;
-    int32_t microsec;
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
+#endif  // defined(__GNUC__)
+    union {
+        struct {
+            uint64_t year:16;
```
Comment (Shylock-Hg) [inline src/common/datatypes/Date.h:138]: The year could be negative.

Comment (dutor) [reply src/common/datatypes/Date.h]: addressed.

**Review by yixinglu — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Review by Shylock-Hg — APPROVED** [review]
LGTM

**Pull request merged**